dev-utils/update-tav-versions.js

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env node
+
+/*
+ * Copyright Elasticsearch B.V. and other contributors where applicable.
+ * Licensed under the BSD 2-Clause License; you may not use this file except in
+ * compliance with the BSD 2-Clause License.
+ */
+
+'use strict';
+
+// Update version property for each module defined in .tav.yml file
+// Usage:
+//      node dev-utils/update-tav-versions.js
+
+const { exec } = require('child_process');
+const { readFileSync, writeFileSync } = require('fs');
+const path = require('path');
+
+const semver = require('semver');
+const yaml = require('js-yaml');
+
+const TOP = path.resolve(path.join(__dirname, '..'));
+const TAV_PATH = path.join(TOP, '.tav.yml');
+
+async function main() {
+  const tavContent = readFileSync(TAV_PATH, { encoding: 'utf-8' });
+  const tavConfig = yaml.load(tavContent);
+  const tavEntries = Object.entries(tavConfig);
+  const modNames = new Set();
+  const versionsMap = new Map();
+
+  const versionModes = {
+    'latest-minors': getLatestMinors,
+  };
+
+  // Fill the module/version map
+  tavEntries.forEach(([name, cfg]) => modNames.add(cfg.name || name));
+
+  // Get versions from all modules
+  await Promise.all(
+    Array.from(modNames).map((name) => {
+      return getModuleVersions(name).then((versions) =>
+        versionsMap.set(name, versions),
+      );
+    }),
+  );
+
+  // Now time to calculate the versions
+  const tavLines = tavContent.split('\n');
+  tavEntries.forEach(([name, cfg]) => {
+    // Array.filter was bogus
+    if (!cfg['versions-selection']) return;
+
+    const { mode, range } = cfg['versions-selection'];
+    const modName = cfg.name || name;
+    const allVers = versionsMap.get(modName);
+    const validVers = allVers.filter((v) => semver.satisfies(v, range));
+    const newVers = versionModes[mode](validVers);
+
+    // TODO: maybe it depends on the mode
+    cfg.versions = newVers.join(' || ');
+
+    // Fill the yamlText wihtout loosing comments and format
+    let verIdx, modIdx;
+
+    for (let i = 0; i < tavLines.length; i++) {
+      const l = tavLines[i];
+      if (!modIdx) {
+        modIdx = l.startsWith(`${name}:`) ? i : undefined;
```

Comment:
Sometimes the name is quoted, e.g. if it includes a `@`:

```yaml
'@koa/router-v8-v11':
  name: '@koa/router'
  versions: '>=8 <11'
  node: '>=8.0.0'
  peerDependencies: koa@2
  commands: node test/instrumentation/modules/koa-router/new-name.test.js
'@koa/router':
  name: '@koa/router'
  versions: '>=11 <13'
  node: '>=12.0.0'
  peerDependencies: koa@2
  commands: node test/instrumentation/modules/koa-router/new-name.test.js

elasticsearch:
  versions: '>=8.0.0'
  commands: node test/instrumentation/modules/elasticsearch.test.js
```

Reply:
thanks, I've changed the code to also cover this case :)